### ISSUE TITLE
[compute_35 Error About CCTag ] 

### ISSUE THREAD

**Issue by Ever-Bright-V** [alicevision/AliceVision#1712]
[ 67%] Built target opencv
[ 67%] Performing build step for 'cctag'
[  1%] Building NVCC (Device) object src/CMakeFiles/CCTag.dir/cctag/cuda/CCTag_generated_triple_point.cu.o
nvcc fatal   : Unsupported gpu architecture 'compute_35'
CMake Error at CCTag_generated_triple_point.cu.o.Release.cmake:220 (message):
  Error generating
  /home/everbright/Codes/aliceVison/build/external/cctag_build/src/CMakeFiles/CCTag.dir/./cctag/cuda/./CCTag_generated_triple_point.cu.o

make[5]: *** [src/CMakeFiles/CCTag.dir/build.make:294：src/CMakeFiles/CCTag.dir/cctag/cuda/CCTag_generated_triple_point.cu.o] 错误 1

**Comment by servantftransperfect** [comment]
Not easy to say without more info, but i think you're using cuda 12+ which is not supported.

**Comment by c-ewing** [comment]
Yes, your GPU is likely to old to use CUDA 12. CUDA 12 requires a compute capability of 5.0+, as per the [Nvidia site](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compute-capabilities). 

See [here](https://developer.nvidia.com/cuda-gpus) for compatible GPUs with compute capability 5.0+

**Comment by BlaineHein** [comment]
The Alicevision code does not always correctly detect the GPU architecture, and then adds in old elements which are not supported by the architecture. Bandaid fix is to remove old architectures from the cctag CMakeLists.txt file. Look for a line that looks like "set(CCTAG_CUDA_CC_LIST_INIT0 3.5 3.7 5.0 5.2)". This would also be fixed by the request to remove EOL GPU from possible choices in the source code. Link to issue #1792
